standards/component-documentation.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Component Documentation
+description: This standard defines how components should be documented in their blueprint description.
+---
+
+# Component Documentation
+
+## Overview
+
+This standard defines how components should be documented in their blueprint description. The goal is to provide a consistent and informative format that works for all components. By following this standard, it will be easier for users to understand how to use a component, because the formatting and terminology is consistent with other components.
+
+## Terminology
+
+- :bearing: **Component**: A logic creation intended to be used inside of other larger logic creations. Examples include ALUs, registers, and memory units.
+- :bearing: **Interface**: A set of logic gates that function as input and/or output gates.
+
+## Blueprint Description
+### Short non-technical description
+The blueprint description should start with a short non-technical description of the component. This should be no longer than a paragraph. This description should explain what the component does, and why it is useful. Avoid immediately diving into technical details, because that's for a later section.
+
+### Interface declaration
+Next, we describe what inputs and outputs the component has. This should be done using the following format:
+`<interface gate color> <amount of gates> <input/output> : <short but descriptive name>`
+(see examples below)
+
+### Detailed description
+After the interface declaration, we should provide a more detailed description of the component. This should include how the component works, and how it should be used. This section should be written in a more technical style, and should be more detailed than the short non-technical description. Here you can use the names of the interfaces you declared earlier.
```

Comment:
- "how the component works" can require a very long description and in some cases it is sufficient to say what it is. 
  e.g. a binary counter doesn't need to be explained much.
- Also, _what the component_ does will be more useful than _how it does this_ to people who want to use the creation, as they can treat it like a black box and still use it.

This can be described in a specific standard for blueprint descriptions of SMLC creations.